kubespawner/spawner.py

```diff
@@ -1486,44 +1527,87 @@ def _validate_image_pull_secrets(self, proposal):
         - `description`: Optional description of this profile displayed to the user.
         - `kubespawner_override`: a dictionary with overrides to apply to the KubeSpawner
           settings. Each value can be either the final value to change or a callable that
-          take the `KubeSpawner` instance as parameter and return the final value.
+          take the `KubeSpawner` instance as parameter and return the final value. This can
+          be further overridden by 'options'
+        ' 'options': A dictionry of sub-options that allow users to further customize the
+          selected profile. Items should have a unique key representing the customization,
+          and the value is a dictionary with the following  keys:
+          - 'display_name': Name used to identify this particular option
+          - 'choices': A dictionary containing list of choices for the user to choose from
+            to set the value for this particular option. The key is an identifier for this
+            choice, and the value is a dictionary with the following possible keys:
+            - 'display_name': Human readable display name for this choice.
+            - 'default': (optional Bool) True if this is the default selected choice
+            - 'kubespawner_override': A dictionary with overrides to apply to the KubeSpawner
+              settings, on top of whatever was applied with the 'kubespawner_override' key
+              for the profile itself. The key should be the name of the kubespawner setting,
+              and value can be either the final value or a callable that returns the final
+              value when called with the spawner instance as the only parameter.
```

Comment:
Perhaps this is a good place to clarify how we combine the two kubespawner_override dictionaries. I understand from the text the option is given precedence over the profile level kubespawner_override, but how in detail?

Situation:

1. profile specific overrides
2. option1 specific overrides based on choice a,b,c
3. option2 specific overrides based on choice x,y,z

Do we start out with profile, followed by overwriting the values of the override dictionaries keys from option1, and then option2? The key question to clarify an answer to in documentation is if there is an override or a merge process.

Reply:
@consideRatio I've updated the text here with the hierarchy of overrides.